src/Shared/ViewBufferTextWriter/MemoryPoolViewBuffer.cs

```diff
@@ -0,0 +1,380 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Html;
+
+/// <summary>
+/// An <see cref="IHtmlContentBuilder"/> that is backed by a buffer provided by <see cref="MemoryPoolViewBufferScope"/>.
+/// </summary>
+[DebuggerDisplay("{DebuggerToString()}")]
+internal sealed class MemoryPoolViewBuffer : IHtmlContentBuilder
```

Comment:
Nit: missing namespace

Reply:
(same suggestion for other files)

Reply:
This was done intentionally so that the source-included files would fall under the same default namespace as the project they were consumed in and no additional usings need to be added.